sklearn/externals/joblib/__init__.py

```diff
@@ -1,133 +1,3 @@
-   persistence and flow-execution logic from domain logic or algorithmic
-   code by writing the operations as a set of steps with well-defined
-   inputs and  outputs: Python functions. Joblib can save their
-   computation to disk and rerun it only if necessary::
-
-      >>> from sklearn.externals.joblib import Memory
-      >>> cachedir = 'your_cache_dir_goes_here'
-      >>> mem = Memory(cachedir)
-      >>> import numpy as np
-      >>> a = np.vander(np.arange(3)).astype(np.float)
-      >>> square = mem.cache(np.square)
-      >>> b = square(a)                                   # doctest: +ELLIPSIS
-      ________________________________________________________________________________
-      [Memory] Calling square...
-      square(array([[0., 0., 1.],
-             [1., 1., 1.],
-             [4., 2., 1.]]))
-      ___________________________________________________________square - 0...s, 0.0min
-
-      >>> c = square(a)
-      >>> # The above call did not trigger an evaluation
-
-2) **Embarrassingly parallel helper:** to make it easy to write readable
-   parallel code and debug it quickly::
-
-      >>> from sklearn.externals.joblib import Parallel, delayed
-      >>> from math import sqrt
-      >>> Parallel(n_jobs=1)(delayed(sqrt)(i**2) for i in range(10))
-      [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0]
-
-
-3) **Fast compressed Persistence**: a replacement for pickle to work
-   efficiently on Python objects containing large data (
-   *joblib.dump* & *joblib.load* ).
-
-..
-    >>> import shutil ; shutil.rmtree(cachedir)
-
-"""
-
-# PEP0440 compatible formatted version, see:
-# https://www.python.org/dev/peps/pep-0440/
-#
-# Generic release markers:
-# X.Y
-# X.Y.Z # For bugfix releases
-#
-# Admissible pre-release markers:
-# X.YaN # Alpha release
-# X.YbN # Beta release
-# X.YrcN # Release Candidate
-# X.Y # Final release
-#
-# Dev branch marker is: 'X.Y.dev' or 'X.Y.devN' where N is an integer.
-# 'X.Y.dev0' is the canonical version of 'X.Y.dev'
-#
-__version__ = '0.13.0'
-
-
-from .memory import Memory, MemorizedResult, register_store_backend
-from .logger import PrintTime
-from .logger import Logger
-from .hashing import hash
-from .numpy_pickle import dump
-from .numpy_pickle import load
-from .compressor import register_compressor
-from .parallel import Parallel
-from .parallel import delayed
-from .parallel import cpu_count
-from .parallel import register_parallel_backend
-from .parallel import parallel_backend
-from .parallel import effective_n_jobs
-
-from .externals.loky import wrap_non_picklable_objects
-
-
-__all__ = ['Memory', 'MemorizedResult', 'PrintTime', 'Logger', 'hash', 'dump',
-           'load', 'Parallel', 'delayed', 'cpu_count', 'effective_n_jobs',
-           'register_parallel_backend', 'parallel_backend',
-           'register_store_backend', 'register_compressor',
-           'wrap_non_picklable_objects']
+from joblib import *
```

Comment:
Also added so that unpickling of existing models doesn't raise errors about non existing paths.

Reply:
Yes, I checked that we can upickle a numpy array and the PCA estimator serialized with joblib in scikit-learn 0.20.2.

Also CircleCI caches pickled datasets used in examples, and I think we could have seen failures in examples if it didn't work.

Without this change I saw failures locally in tests that used the cached data for `fetch_20newsgroups_vectorized`

Reply:
You won't see circle ci examples fail unless you force a full build

Reply:
Right, forgot about that. Re-triggered the full CircleCI build -- no errors still.

Reply:
We can keep a small sklearn estimator serialized with joblib in scikit-learn 0.20.2 in a test directory, and include a test that unpickles it to make sure it doesn't error.

Reply:
@thomasjpfan There is a test for unpickling numpy arrays in https://github.com/scikit-learn/scikit-learn/pull/13531/files#r269922098 I'm not convinced that unpickling an actual scikit-learn estimator as part of the test suite is worth it: if it's simple we would be basically testing the python cPickle module. If it's more complex and interacts with lots of parts of scikit-learn it is bound to break as our estimators evolve (and we won't be able to unpickle it).

I have tested that unpickling works outside of the test suite as well as mentioned above.